src/ui/public/vis/agg_configs.js

```diff
@@ -145,14 +144,41 @@ export default function AggConfigsFactory(Private) {
     });
 
     removeParentAggs(dslTopLvl);
-
     return dslTopLvl;
   };
 
   AggConfigs.prototype.getRequestAggs = function () {
-    return _.sortBy(this, function (agg) {
+
+
+    const aggregations = _.sortBy(this, function (agg) {
```

Comment:
There's currently an extension point in the AppConfig to allow each individual item to customize the responseAggs, I added one for the requestAggs as well so that we can do the geohash logic there and it seems to work: https://github.com/kobelb/kibana/commit/fd6b34495cb41d50e9820018bd22e76d11833565

That way we don't have to hard-code the logic for specific aggregations here.

We should add unit-tests for this, regardless of which direction we go.